python/res/job_queue/workflow_runner.py

```diff
@@ -41,9 +54,18 @@ def isCancelled(self):
     def cancel(self):
         if self.isRunning():
             self.__workflow.cancel()
+        self.wait()
+
+    @property
+    def runner_thread(self):
+        return self.__workflow_thread
 
     def wait(self):
-        self.__workflow.wait()
+        if self.runner_thread:
+            self.runner_thread.join()
+        print('DBG wait finnished with following exception', self.runner_thread._exception)
```

Comment:
what happens if **not** self.runner_thread ? from what i understand your print will fail when calling _exception on it?

Reply:
Good point, should be inside the if statement